lib/OperationHandler.cpp

```diff
@@ -398,22 +450,27 @@ ScanResponse OperationHandler::HandleScanRequestSerialized(const ScanRequest& sc
     const auto timeout = std::chrono::steady_clock::now() + std::chrono::seconds(10);
     for (auto& scanFuture : futureScanResults)
     {
-        std::vector<ScannedBss> scanResults;
-        if (scanFuture.wait_until(timeout) != std::future_status::ready)
-        {
-            LOG_WIN32_MSG(ERROR_TIMEOUT, "Scan timed out.");
-        }
-        else
+        try
         {
-            scanResults = scanFuture.get();
-        }
+            std::vector<ScannedBss> scanResults;
+            if (scanFuture.wait_until(timeout) != std::future_status::ready)
```

Comment:
Sorry, I know this is existing code, but should these maybe we waited on in an async manner in parralel instead? With the current method, if an early future times out, all the others will back up behind it, even if they've completed.

I know this would be non-trivial to implement, but I'm wondering if it would help performance. I'm also not 100% familliar with this code path, so maybe this suggestion doesn't make complete sense. Please school me :)

Reply:
This is already an async wait, since it is using `wait_until` and not `wait_for`. If a future timeout, all other are processed without waiting more.

Reply:
Hmm, I thought `wait_until` blocks for the specified timeout period. I'll look at the docs.

Reply:
Resolving for now since it's not directly related to this PR. Please let me know or open an issue if you think this is incorrect.